lib/pymedphys/_experimental/streamlit/apps/sum_doses.py

```diff
@@ -0,0 +1,319 @@
+from pymedphys._dicom.dose import dose_from_dataset
+from pymedphys._streamlit import categories
+
+CATEGORY = categories.PRE_ALPHA
+TITLE = "Sum Coincident DICOM Doses"
+
+HERE = pathlib.Path(__file__).parent.resolve()
+
+
+def _check_files_valid(
+    files: Sequence[pathlib.Path],
+) -> Union["List[pydicom.dataset.Dataset]", None]:
+
+    if not len(files) >= 2:
+        raise ValueError("`files` must contain at least 2 elements")
+
+    datasets = []
+    ds0 = None
+
+    for i, fh in enumerate(files):
+        try:
+            ds = pydicom.dcmread(fh)
+        except pydicom.errors.InvalidDicomError:
+            st.error(f"'{fh.name}' is not a valid DICOM file")
+            return None
+
+        if not ds.Modality == "RTDOSE":
+            st.error(f"File '{fh.name}' is not a valid DICOM RT Dose file")
+            return None
+
+        if i == 0:
+            ds0 = ds
+        else:
+            if not ds.PatientID == ds0.PatientID:
+                st.error(
+                    f"File '{fh.name}' has a different DICOM Patient "
+                    f"ID from '{files[0].name}'"
+                )
+                return None
+            if not ds.DoseUnits == ds0.DoseUnits:
+                st.error(
+                    f"File '{fh.name}' has a different value for "
+                    f"DoseUnits ({ds.DoseUnits}) from "
+                    f"'{files[0].name}' ({ds0.DoseUnits})"
+                )
+                return None
+
+        if not ds.DoseSummationType == "PLAN":
+            st.error(f"File {fh.name} is not a 'plan' dose")
+            return None
+
+        datasets.append(ds)
+
+    return datasets
+
+
+def _save_dataset_to_downloads_dir(ds: "pydicom.dataset.Dataset"):
+    STREAMLIT_STATIC_PATH = pathlib.Path(st.__path__[0]) / "static"
+
+    DOWNLOADS_PATH = STREAMLIT_STATIC_PATH / "downloads"
+    DOWNLOADS_PATH.mkdir(parents=True, exist_ok=True)
+
+    ds.save_as(DOWNLOADS_PATH / "RD.Summed.dcm")
+
+
+def coords_in_datasets_are_equal(datasets: "Sequence[pydicom.dataset.Dataset]") -> bool:
+    """True if all DICOM datasets have perfectly matching coordinates
+
+    Parameters
+    ----------
+    datasets : sequence of pydicom.dataset.Dataset
+        A sequence of DICOM datasets whose coordinates are to be
+        compared.
+
+    Returns
+    -------
+    bool
+        True if coordinates match for all datasets, False otherwise.
+    """
+
+    if not len(datasets) >= 2:
+        raise ValueError("At least two datasets must be provided for comparison")
+
+    # Quick shape (sanity) check
+    if not all(
+        ds.pixel_array.shape == datasets[0].pixel_array.shape for ds in datasets
+    ):
+        return False
+
+    # Full coord check:
+    all_concat_axes = [np.concatenate(xyz_axes_from_dataset(ds)) for ds in datasets]
+
+    return all(np.allclose(a, all_concat_axes[0]) for a in all_concat_axes)
+
+
+def get_pretty_patient_name_from_dicom_dataset(
```

Comment:
@SimonBiggs, @sjswerdloff - a handy addition to the library? Does one already exist?

Reply:
Yup, that's the first thing I thought when I saw that. "Straight to the pool room" for that one 🙂